frontend/src/metabase/App.jsx

```diff
@@ -134,13 +142,58 @@ export default class App extends Component {
             />
           </SearchBarContent>
         </SearchBarContainer>
-        <ProfileLinkContainer>
-          <ProfileLink {...this.props} user={currentUser} />
-        </ProfileLinkContainer>
+        <NewButton setModal={this.setModal.bind(this)} />
       </AppBar>
     );
   };
 
+  setModal(modal) {
+    this.setState({ modal });
+    if (this._newPopover) {
+      this._newPopover.close();
+    }
+  }
+
+  renderModalContent() {
+    const { modal } = this.state;
+    const { onChangeLocation } = this.props;
+
+    switch (modal) {
+      case MODAL_NEW_COLLECTION:
+        return (
+          <CollectionCreate
+            onClose={() => this.setState({ modal: null })}
+            onSaved={collection => {
+              this.setState({ modal: null });
+              onChangeLocation(Urls.collection(collection));
+            }}
+          />
+        );
+      case MODAL_NEW_DASHBOARD:
+        return (
+          <CreateDashboardModal
+            onClose={() => this.setState({ modal: null })}
+          />
+        );
+      default:
+        return null;
+    }
+  }
+
+  renderModal() {
+    const { modal } = this.state;
+
+    if (modal) {
+      return (
+        <Modal onClose={() => this.setState({ modal: null })}>
```

Comment:
I guess we should extract `closeModal` method as it's going to be used for every new modal